src/adapter/zboss/adapter/zbossAdapter.ts

```diff
@@ -54,56 +58,45 @@ export class ZBOSSAdapter extends Adapter {
 
     private async processMessage(frame: ZBOSSFrame): Promise<void> {
         logger.debug(() => `processMessage: ${JSON.stringify(frame)}`, NS);
-        if (
-            frame.type == FrameType.INDICATION &&
-            frame.commandId == CommandId.ZDO_DEV_UPDATE_IND &&
-            frame.payload.status == DeviceUpdateStatus.LEFT
-        ) {
-            logger.debug(`Device left network request received: ${frame.payload.nwk} ${frame.payload.ieee}`, NS);
-            const payload: DeviceLeavePayload = {
-                networkAddress: frame.payload.nwk,
-                ieeeAddr: frame.payload.ieee,
-            };
-
-            this.emit('deviceLeave', payload);
-        }
-        if (frame.type == FrameType.INDICATION && frame.commandId == CommandId.NWK_LEAVE_IND) {
-            logger.debug(`Device left network request received from ${frame.payload.ieee}`, NS);
-            const payload: DeviceLeavePayload = {
-                networkAddress: frame.payload.nwk,
-                ieeeAddr: frame.payload.ieee,
-            };
-
-            this.emit('deviceLeave', payload);
-        }
-        if (frame.type == FrameType.INDICATION && frame.commandId == CommandId.ZDO_DEV_ANNCE_IND) {
-            logger.debug(`Device join request received: ${frame.payload.nwk} ${frame.payload.ieee}`, NS);
-            const payload: DeviceJoinedPayload = {
-                networkAddress: frame.payload.nwk,
-                ieeeAddr: frame.payload.ieee,
-            };
-
-            this.emit('deviceJoined', payload);
-        }
```

Comment:
the device join handler is forgotten. if we do not process ZDO_DEV_ANNCE_IND, then we will process ZDO_DEV_AUTHORIZED_IND
```
            } else if (frame.commandId == CommandId.ZDO_DEV_AUTHORIZED_IND) {
                const payload: DeviceJoinedPayload = {
                    networkAddress: frame.payload.nwk,
                    ieeeAddr: frame.payload.ieee,
                };

                this.emit('deviceJoined', payload);
```

Reply:
I made a few tweaks according to:
https://github.com/espressif/esp-zboss-lib/blob/2abb29f3a1934ac48bf58b2245dc18906359c97d/include/zboss_api_zdo.h#L1237-L1371
https://developer.nordicsemi.com/nRF_Connect_SDK/doc/zboss/3.11.2.1/structzb__zdo__signal__device__update__params__s.html
https://developer.nordicsemi.com/nRF_Connect_SDK/doc/zboss/3.11.2.1/structzb__zdo__signal__device__authorized__params__s.html
https://developer.nordicsemi.com/nRF_Connect_SDK/doc/zboss/3.11.2.1/group__zb__comm__signals.html#zdo_authorization_type

Should fit better?
_Looks a lot like EmberZNet logic._

Reply:
Sorry. it's bad. There is a different frame structure here
https://cloud.dsr-corporation.com/index.php/s/BAn4LtRWbJjFiAm
![image](https://github.com/user-attachments/assets/eb7750ec-ab55-4f3f-8a5a-b3ac62ae1abe)

Reply:
```
zh:zboss:uart: <-- FRAME: 00021502656a8a19004b1202698601
zh:zboss: processMessage: {"version":0,"type":2,"commandId":533,"tsn":0,"payload":{"ieee":"0x02124b00198a6a65","nwk":38088,"status":1}}
zh:zboss: Device 0x02124b00198a6a65:38088 UNSECURED_JOIN undefined via undefined.
zh:zboss: Device 0x02124b00198a6a65:38088 was denied joining via undefined (undefined).
```

Reply:
Yikes! So nordic has the full spec on their website, but it doesn't match what they have in their devices? 😢
Is it an outdated version maybe? Seems strange that espressif would implement it that way too.

I reverted the changes. I'll let you deal with completing the implementation in future PRs 😛